test/nn/encoder/test_stype_encoder.py

```diff
@@ -426,3 +432,148 @@ def test_text_tokenized_encoder():
         assert torch.allclose(
             feat_text[key].offset,
             tensor_frame.feat_dict[stype.text_tokenized][key].offset)
+
+
+def test_linear_model_encoder():
+    num_rows = 20
+    out_channels = 8
+    data_stypes = [
+        torch_frame.numerical,
+        torch_frame.text_embedded,
+        torch_frame.timestamp,
+        torch_frame.categorical,
+        torch_frame.multicategorical,
+        torch_frame.sequence_numerical,
+        torch_frame.embedding,
+    ]
+    dataset = FakeDataset(
+        num_rows=num_rows,
+        stypes=data_stypes,
+        col_to_text_embedder_cfg=TextEmbedderConfig(
+            text_embedder=HashTextEmbedder(out_channels=out_channels),
+            batch_size=None,
+        ),
+    )
+    dataset.materialize()
+    tensor_frame = dataset.tensor_frame
+    stats_list = []
+    col_to_model_cfg = {}
+    encoder_dict = {}
+    for data_stype in data_stypes:
+        data_stype = data_stype.parent
+        stats_list.extend(
+            dataset.col_stats[col_name]
+            for col_name in tensor_frame.col_names_dict[data_stype])
+        for col_name in tensor_frame.col_names_dict[data_stype]:
+            if data_stype == torch_frame.embedding:
+                in_channels = dataset.col_stats[col_name][StatType.EMB_DIM]
+                model = EmbeddingModel(in_channels, out_channels)
+            elif data_stype == torch_frame.numerical:
+                in_channels = 1
+                model = NumericalModel(in_channels, out_channels)
+            elif data_stype == torch_frame.timestamp:
+                in_channels = 7
```

Comment:
same. no need to support `in_channels`. Also, would be nicer to not hard-code `7` in the code.